test/azure/RspecTests/parameters_constraints_spec.rb

```diff
@@ -0,0 +1,211 @@
+# encoding: utf-8
```

Comment:
should the tests for constraints be under the "vanilla" folder?

Reply:
As I see it, the vanilla folder is for the basic tests such array_spec, bool_spec, etc. I think this should stay under azure folder. Correct me if I am wrong.

Reply:
The tests are divided depending on the flavor of generator being used, it depends on autorest using the vanilla generator or the azure flavor of the generator. 
Since the constraint changes are in the vanilla code, it means that any generated code, not necessarily the one generated with azure flavor (--azure-arm=true), would have to honor them, that's the reason why I think the tests correspond to the vanilla folder.